sdk/monitor/monitor-opentelemetry-exporter/src/utils/spanUtils.ts

```diff
@@ -95,102 +115,148 @@ function createPropertiesFromSpan(span: ReadableSpan): [Properties, Measurements
-    resultCode: String(span.status.code),
-    target: span.attributes[HTTP_URL] as string | undefined,
+    success: span.status.code != SpanStatusCode.ERROR,
+    resultCode: "0",
     type: "Dependency",
     duration: msToTimeSpan(hrTimeToMilliseconds(span.duration)),
-    version: 1
+    version: 2
   };
-
-  if (span.attributes[HTTP_STATUS_CODE]) {
-    data.type = "HTTP";
-    data.resultCode = String(span.attributes[HTTP_STATUS_CODE]);
-  }
-
-  if (span.attributes[GRPC_STATUS_CODE] !== undefined) {
-    data.type = "GRPC";
-    data.resultCode = String(span.attributes[GRPC_STATUS_CODE]);
-  }
-
-  if (span.attributes[GRPC_METHOD]) {
-    data.target = String(span.attributes[GRPC_METHOD]);
-    data.data = String(span.attributes[GRPC_METHOD]);
+  const httpMethod = span.attributes[SemanticAttributes.HTTP_METHOD];
+  const dbSystem = span.attributes[SemanticAttributes.DB_SYSTEM];
+  const rpcMethod = span.attributes[SemanticAttributes.RPC_METHOD];
+  // HTTP Dependency
+  if (httpMethod) {
+    remoteDependencyData.type = DependencyTypes.Http;
+    remoteDependencyData.data = getUrl(span);
+    const httpStatusCode = span.attributes[SemanticAttributes.HTTP_STATUS_CODE];
+    if (httpStatusCode) {
+      remoteDependencyData.resultCode = String(httpStatusCode);
+    }
+    let target = getTarget(span);
+    if (target) {
+      let url = new URL(target);
+      // Ignore port if it is a default port
+      if (url.port === "80" || url.port === "443") {
+        url.port = "";
+      }
+      remoteDependencyData.target = `${url}`;
+    }
   }
-
-  if (span.attributes[HTTP_URL]) {
-    const url = new URL(span.attributes[HTTP_URL] as string);
-    data.target = url.hostname;
-    data.data = url.href;
-
-    if (span.attributes[HTTP_METHOD]) {
-      data.name = `${span.attributes[HTTP_METHOD] as string} ${url.pathname}`;
+  // DB Dependency
+  else if (dbSystem) {
+    remoteDependencyData.type = String(dbSystem);
+    const dbStatement = span.attributes[SemanticAttributes.DB_STATEMENT];
+    if (dbStatement) {
+      remoteDependencyData.data = String(dbStatement);
+    }
+    let target = getTarget(span);
+    const dbName = span.attributes[SemanticAttributes.DB_NAME];
+    if (target) {
+      remoteDependencyData.target = dbName ? `${target}/${dbName}` : `${target}`;
+    } else {
+      remoteDependencyData.target = dbName ? `${dbName}` : `${dbSystem}`;
     }
   }
-
-  if (span.attributes[DB_STATEMENT]) {
-    data.name = String(span.attributes[DB_STATEMENT]);
-    data.data = String(span.attributes[DB_STATEMENT]);
-    data.type = "DB";
-    if (span.attributes[DB_NAME]) {
-      data.target = String(span.attributes[DB_NAME]);
+  // grpc Dependency
+  else if (rpcMethod) {
+    const grpcStatusCode = span.attributes[SemanticAttributes.RPC_GRPC_STATUS_CODE];
+    if (grpcStatusCode) {
+      remoteDependencyData.resultCode = String(grpcStatusCode);
+    }
+    let target = getTarget(span);
+    if (target) {
```

Comment:
Is this logic for `target` specified for `rpc` yet?

Reply:
In case of rpc:

Derive target from peerService or net* attributes as above.
Default: If none of the above are available, assign rpcSystem to target.

https://github.com/microsoft/common-schema/blob/main/Mappings/AzureMonitor-AI.md#remotedependencytarget